docs/TRANSLATING.md

```diff
@@ -1,80 +1,289 @@
+- **Poedit**, cross-platform
+- **Lokalize**, a KDE tool for Linux
+- **Gtranslator**, a GNOME tool for Linux
+- many web-based tools
+
+For DOSBox Staging translation files, it is best to configure the editor to use
+a monospaced (non-proportional) font:
+
+- **Poedit**\
+_File_ -> _Preferences_ -> _General_ -> _Appearance_ ->
+_Use&nbsp;custom&nbsp;text&nbsp;field&nbsp;font_
+- **Lokalize**\
+_Settings_ -> _Configure&nbsp;Lokalize..._ -> _Appearance_ ->
+_Editor&nbsp;font_ -> _Show&nbsp;only&nbsp;monospaces&nbsp;fonts_
+- **Gtranslator**\
+hamburger&nbsp;menu -> _Preferences_ -> _Editor_ -> _Font_
+
+## Adding new translation
+
+To start a new translation, execute the following commands inside the current
+alpha build of DOSBox Staging:
+
+```bat
+config -set language=en
+config -wl xx.po
 ```
-dosbox -lang fr -c "config -wl fr.lng" -c "exit"
+
+where _xx_ is a language code. If only possible, use a two-letter code
+according to the [ISO 639](https://en.wikipedia.org/wiki/List_of_ISO_639_language_codes)
+standard.
+
+This will create a `xx.po` file containing all the English messages, ready for
+translating.
+
+## Updating existing translation
+
+Before you start editing any current `PO` file, you need to refresh its
+content. Execute the following commands inside the current alpha build of
+DOSBox Staging:
+
+```bat
+config -set language=xx
+config -wl xx.po
 ```
 
-or, if you prefer to save the file from inside DOSBox Staging:
+where _xx_ is a language code.
 
+This will create an updated `xx.po` file with removed outdated messages,
+updated English messages, updated source code references, outdated translations
+marked accordingly, and without messages that were removed from the source
+code. This newly created file is ready for updating the translation.
+
+## PO file format
+
+The `PO` file format is described in detail on the
+[gettext documentation](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html)
+web page; this chapter only provides a brief introduction.
+
+The file consists of entries of roughly the same layout, separated with empty
+lines. DOSBox Staging needs the file to be saved as UTF-8 — don't change the
+encoding, otherwise the file won't work correctly.
+
+### Standard metadata entry
+
+The first entry in the file is reserved for the standard gettext metadata — it
+always contains an empty English string (`msgid`); all the metadata entries are
+put in a translated string (`msgstr`).
+
+```po
+msgid ""
+msgstr ""
+"Project-Id-Version: dosbox-staging\n"
+"Report-Msgid-Bugs-To: https://github.com/dosbox-staging/dosbox-staging/issues\n"
+...
 ```
-config -set language=fr
-config -wl fr.lng
+
+If you are editing the file by hand, using an ASCII editor, don't change this
+entry.
+
+### DOSBox Staging metadata entries
+
+DOSBox Staging needs some additional, non-standard metadata — it can be
+recognized by `#METADATA` keyword present in both the context (`msgctxt`) and
```

Comment:
"non-standard metadata marked with the `#METADATA` keyword in both the..."

Reply:
OK, changed.